rocketpy/Rocket.py

```diff
@@ -278,8 +309,138 @@ def __init__(
         # Evaluate static margin (even though no aerodynamic surfaces are present yet)
         self.evaluateStaticMargin()
 
+        self.evaluateDragCoefficient()
+
         return None
 
+    def evaluateDragCoefficient(self):
+        """Calculates and returns the drag coefficient as the sum of
+        of the individual drag coefficients at zero angle of attack
+        as a function of the Mach number. The angle of attack is
+        assumed to be zero.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        self.viscous_friction_coefficient : Function
+            Function of mach number expressing the viscous friction coefficient,
+            defined as in the literature.
+        """
+        self.evaluateViscousFrictionCoefficient()
+        self.evaluateForebodyDragCoefficient()
+
+        self.drag_coefficient_estimate = self.forebody_drag_coefficient + np.sum(
+            [
+                self.aerodynamicSurfaces[i][2]
+                for i in range(len(self.aerodynamicSurfaces))
+            ]
+        )
+
+        if not self.power_off_drag_as_input:
+            self.powerOffDrag = self.drag_coefficient_estimate
+        if not self.power_on_drag_as_input:
+            self.powerOnDrag = self.drag_coefficient_estimate
+
+        return self.drag_coefficient_estimate
+
+    def evaluateViscousFrictionCoefficient(self):
+        """Calculates and returns the viscous friction coefficient as
+        a function of the Mach number. The viscous friction coefficient
+        depends on the Reynolds number and given by a mathematical
+        expression. Some assumptions for calculating this coefficient
+        are that the angle of attack equals zero and the speed of sound
+        is constant and equal to the value passed in init.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        self.viscous_friction_coefficient : Function
+            Function of mach number expressing the viscous friction coefficient,
+            defined as in the literature.
+        """
+
+        def calculations(mach):
+            reynolds = (
+                (self.air_density ** 2)
+                * self.speed_of_sound
+                * self.rocket_length
```

Comment:
Is there a specific reason for using rocket_length as the reference length instead of diameter? I believe different literatures has different opinons on that, but I'm not an expert tough

Reply:
The literature we are following explicitly states recommends that the reference length to be used is the rocket length. If you want to take a look, we were following the document "Estimating the dynamic and aerodynamic paramters of passively controlled high power rockets for flight simulaton", by Simon Box, Christopher M. Bishop and Hugh Hunt. The document can also be found in the following link http://cambridgerocket.sourceforge.net/AerodynamicCoefficients.pdf

Reply:
I agree using length makes more sense in this case, since we are interested in the turbulent behavior along the rocket axis, rather than relative to its diameter.